### DISCUSSION TITLE
Monorepo: Fails communicating with backend once deployed in Vercel

### DISCUSSION THREAD

**Discussion by amine-abdelli** [vercel/vercel#8714] [Help]
Hi everyone !

I'm trying to deploy a turborepo monorepo in Vercel.
```
root/
├── package.json
├── node_modules/
├── turbo.json
└── packages/
      ├── client/                  ( Next.js running on port :3000 )
      ├── server1/               ( Node/Express server running on port :4000 )
      └── server2/               ( Socket.io server running on port :4001 )
```

In development i just run `yarn start` at the root of my project and it will run my frontend, server 1 and server 2. My frontend can then communicate with my servers via localhost:4000 and localhost:4001. 

However once my project is deployed in Vercel i have no idea of what urls i should tell my frontend to communicate to. I naively tried https://my-project.vercell.app:4000 and https://my-project.vercell.app:4001 but obviously it didn't work 😅 ! So what do you suggest me to try to make everything communicate together ? Any suggestion ? 

So far i'm building the app then run this build and the frontend runs well. 

Thanks a lot for reading this post !

Cheers !

Amine

**Comment by amyegan** [comment]
Hi @amine-abdelli! We have some guides to help you deploy a monorepo on Vercel. You should take a look at these ["Monorepos"](https://vercel.com/docs/concepts/monorepos) and ["Turborepo"](https://vercel.com/docs/concepts/monorepos/turborepo#) docs. 

I believe you need to create a separate project for each app in the monorepo. So you would have something like https://my-project.vercell.app, https://my-project-server1.vercell.app, and https://my-project-server2.vercell.app. You can use [rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) to get everything to appear like it's at the same domain.
